### PR TITLE
website: upgrade docusaurus to 2.4.0

### PR DESCRIPTION
**Pull request by sluongng**
This is a follow up of #3892.

Several changes needed to be made to accomodate for the new version:
- Deprecate colorMode.switchConfig config
- Use `className` instead of `class` in blog post's nav component.
- Several component deprecation / replacements
- BlogPostItem needed to be wrapped around by BlogPostProvider
- Tag struct in each blog post was also updated

**Related issues**: N/A

### CODE REVIEW

**Comment by sluongng** [comment]
I have tested this locally and found everything to be working quite well.

The PR could use some cleanup(?) but I would love to get 1 round of review in first.

**Comment by siggisim** [comment]
I've attempted this upgrade before, but ran into a bunch of small issues. I've committed a bunch of CSS changes to this PR that should fix a lot of them

**Review by siggisim — APPROVED** [review]

File: website/theme/BlogTagsPostsPage/index.tsx [hunk 1]
```diff
@@ -56,14 +54,17 @@ function BlogTagsPostPage(props: Props): JSX.Element {
...
                   assets={BlogPostContent.assets}
                   metadata={BlogPostContent.metadata}
+                  content={BlogPostContent}
                   truncated>
-                  <BlogPostContent />
-                </BlogPostItem>
+                  <BlogPostItem truncated>
+                    <BlogPostContent />
+                  </BlogPostItem>
+                </BlogPostProvider>
```
Comment (siggisim) [inline website/theme/BlogTagsPostsPage/index.tsx:67]: I think we somehow lost the "Read more" button here (you can see it on the prod `/blog/` page

**Review by siggisim — APPROVED** [review]

File: website/theme/BlogTagsPostsPage/index.tsx (hunk 1 above)
Comment (sluongng) [reply website/theme/BlogTagsPostsPage/index.tsx:67]: Fixed that, it was `s/metadata.truncated/metadata.hasTruncateMarker/`

**Pull request merged**